types/schema.d.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Schemas in Fastify follow the JSON-Schema standard. For this reason
+ * we have opted to not ship strict schema based types. Instead we provide
+ * an example in our documentation on how to solve this problem. Check it
+ * out here:
+ */
+export type FastifySchema = {
+  body?: unknown,
+  querystring?: unknown,
+  params?: unknown,
+  headers?: unknown
+};
+
+/**
+ * Compiler for FastifySchema Type
+ */
+export type FastifySchemaCompiler = (schema: FastifySchema) => void // todo
```

Comment:
TODO?

Reply:
Yes the previous types simply have this as `Function` instead of `void`. I'm going to change this to `unknown` so the user can still use it, but will know it is not typed yet.

Reply:
This also will be fixed with the work happening in #1668 schemas are very tough to type so I don't want to hold up the rest of the PR for just it.